userspace/libsinsp/test/classes/sinsp_observer.cpp

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+Copyright (C) 2025 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include <sinsp_with_test_input.h>
+#include "sinsp_observer.h"
+
+class test_observer : public sinsp_observer {
+public:
+	test_observer():
+	        m_clone_ctr(0),
+	        m_execve_ctr(0),
+	        m_open_ctr(0),
+	        m_read_ctr(0),
+	        m_close_ctr(0) {}
+
+	void on_read(sinsp_evt* evt,
+	             int64_t tid,
+	             int64_t fd,
+	             sinsp_fdinfo* fdinfo,
+	             const char* data,
+	             uint32_t original_len,
+	             uint32_t len) override {
+		std::cerr << "[          ] read = " << evt->get_name() << " fd " << fdinfo->m_fd << " data "
+		          << data << std::endl;
```

Comment:
I know it's a bit more work, but instead of printing out text could we save this into a field and then `expect` or `assert` that the saved string is right?

Reply:
Right, done!